modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/session.js

```diff
@@ -277,33 +286,50 @@ AUI.add(
 						var warningMoment = false;
 
 						var hasExpired = elapsed >= sessionLength;
+						var hasExpiredTimeoutOffset =
+							elapsed >= sessionLength - sessionTimeoutOffset;
 						var hasWarned = elapsed >= warningTime;
 
-						if (hasWarned) {
+						if (hasExpiredTimeoutOffset || hasWarned) {
 							if (timestamp == 'expired') {
 								expirationMoment = true;
 								extend = false;
 								hasExpired = true;
+								hasExpiredTimeoutOffset = true;
 							}
 
-							if (hasExpired && sessionState != 'expired') {
-								if (extend) {
+							if (
+								hasExpiredTimeoutOffset &&
+								sessionState != 'expired'
+							) {
+								if (extend && !hasExpired) {
 									expirationMoment = false;
 									hasExpired = false;
+									hasExpiredTimeoutOffset = false;
 									hasWarned = false;
 									warningMoment = false;
 
 									instance.extend();
 								}
 								else {
+									if (extend && hasExpired) {
+										// eslint-disable-next-line no-console
```

Comment:
Hey Jorge, We prefer to not use `console.log` and you can see the explanation [here](https://github.com/liferay/liferay-frontend-projects/blob/dc7d947da927de33e556c7570c007325efa560dc/guidelines/dxp/debugging.md#on-the-use-of-consolelog). 

In addition, If We allow ppl from using console.log in their components it can be very difficult to manage.

Also, from [this doc](https://github.com/liferay/liferay-frontend-projects/blob/dc7d947da927de33e556c7570c007325efa560dc/guidelines/dxp/debugging.md#on-the-use-of-consolelog),

>  always prefer to implement graceful recovery and user experiences that are designed with resilience in mind.

[We're doing it](https://github.com/liferay-frontend/liferay-portal/blob/8a882638533bd8f10bc25b01335f28c27c7fc1ac/modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/session.js#L623), using `openToast` to alert users the session expired

Reply:
Hi @diegovs

No problem, we can remove that console.log:
- The user is already warned in the browser with a error popup message by the instance.expire() call.
- I added the console.log call just to be able to do a better diagnose in case a customer has problems with the session extension, printing the related variables, but we could safely remove it.

I will update the PR tomorrow, removing this.

Let me know if you need more information.

Reply:
Hi @diegonvs 

I have added a new commit to the PR that removes the console.log see: [`4b47f08` (#1349)](https://github.com/liferay-frontend/liferay-portal/pull/1349/commits/4b47f085a276e06912cacc28c6f6dfc88319bea1)